node/src/store/sled_store.rs

```diff
@@ -0,0 +1,497 @@
+use std::convert::Infallible;
+use std::io;
+use std::ops::Deref;
+use std::path::Path;
+
+use async_trait::async_trait;
+use celestia_types::hash::Hash;
+use celestia_types::ExtendedHeader;
+use directories::ProjectDirs;
+use sled::transaction::{ConflictableTransactionError, TransactionError};
+use sled::{Db, Error as SledError, Transactional, Tree};
+use tempdir::TempDir;
+use tendermint_proto::Protobuf;
+use tracing::debug;
+
+use crate::store::Store;
+use crate::store::{Result, StoreError};
+
+const HEAD_HEIGHT_KEY: &[u8] = b"KEY.HEAD_HEIGHT";
+const HASH_TREE_ID: &[u8] = b"HASH";
+const HEIGHT_TO_HASH_TREE_ID: &[u8] = b"HEIGHT";
+
+#[derive(Debug)]
+pub struct SledStore {
+    db: Db,
+    headers: Tree,
+    height_to_hash: Tree,
+}
+
+impl SledStore {
+    pub fn new(network_id: &str) -> sled::Result<Self> {
+        let Some(project_dirs) = ProjectDirs::from("co", "eiger", "celestia") else {
+            return Err(sled::Error::Io(io::Error::new(
+                io::ErrorKind::Other,
+                "Failed to get cache directory",
+            )));
+        };
+        let mut db_path = project_dirs.cache_dir().to_owned();
+        db_path.push(network_id);
+
+        Self::new_in_path(db_path)
+    }
+
+    pub fn new_temp() -> sled::Result<Self> {
+        let tmp_path = TempDir::new("celestia")?.into_path();
+
+        let db = sled::Config::default()
+            .path(tmp_path)
+            .temporary(true)
```

Comment:
Does `temporary` cleans up the directory in drop? If not, then we need to do it, or keep the `TempDir` in the struct, which does this.

Reply:
Yes, sled cleans up the directory: https://docs.rs/sled/latest/sled/struct.Config.html#method.temporary